zeroconf/__init__.py

```diff
@@ -1073,44 +1082,96 @@ def write_record(self, record: DNSRecord, now: float) -> int:
         # Here is the short we adjusted for
         self.insert_short(index, length)
 
+        len_limit = _MAX_MSG_ABSOLUTE if allow_long else _MAX_MSG_TYPICAL
+
         # if we go over, then rollback and quit
-        if self.size > _MAX_MSG_ABSOLUTE:
+        if self.size > len_limit:
             while len(self.data) > start_data_length:
                 self.data.pop()
             self.size = start_size
-            self.state = self.State.finished
-            return 1
-        return 0
+            return False
+        return True
 
     def packet(self) -> bytes:
-        """Returns a string containing the packet's bytes
+        """Returns a string containing the first packet's bytes.
+
+        Generally, you want to use packets() in case the response
+        does not fit in a single packet, but this exists for
+        backward compatibility."""
+        p = self.packets()
+        if len(p) > 0:
+            if len(p[0]) > _MAX_MSG_ABSOLUTE:
+                self.log_warning_once("Created over-sized packet (%d bytes) %r", len(p[0]), p[0])
+            return p[0]
+        else:
+            return b''
+
+    def packets(self) -> List[bytes]:
+        """Returns a list of strings containing the packets' bytes
```

Comment:
"bytestrings" instead of "strings"? "strings" implies text.

Reply:
Sure.  I think I was just copying the former docstring language, but this is a good idea to be clear about.